compiler/src/dotty/tools/dotc/transform/Splicing.scala

```diff
@@ -198,10 +200,57 @@ class Splicing extends MacroTransform:
       val newTree = transform(tree)
       val (refs, bindings) = refBindingMap.values.toList.unzip
       val bindingsTypes = bindings.map(_.termRef.widenTermRefExpr)
-      val methType = MethodType(bindingsTypes, newTree.tpe)
```

Comment:
The code example in the doc comment above SpliceTransformer also needs to be updated to use a polymorphic function.